tools/OMIcheck/OMIcheck.ps1

```diff
@@ -0,0 +1,73 @@
+﻿$upgradeOMI = $false; #detect only
+$checkScriptPath  = "C:\linux\OMIcheck\omi_check.sh"
+$upgradeScriptPath = "C:\linux\OMIcheck\omi_upgrade.sh"
+
+# Get all Azure Subscriptions
+$subs = Get-AzSubscription
+ 
+# For each subscription execute to find (and upgrade) omi patch on Linux VMs
+foreach ($sub in $subs)
+{
+    #DEBUG: Limit to one subscription only.
+    #if ($sub.Name -eq "Geneva Monitoring Agent - LinuxMdsd")
+    #{
+        # Set Azure Subscription context    
+        Set-AzContext -Subscription $sub.Id
+        
+        Write-Output "Listing Virutal Machines in subscription '$($sub.Name)'"
+        $VMs = Get-AzVM -Status
+        $VMsSorted = $VMs | Sort-Object -Property ResourceGroupName
+        $PreviousRG = ""
+        foreach($VM in $VMs)
+        {
+            #DEBUG: limit to 1 vm only
+            #if ($vm.name -ne "my-testing-vm")
+            #{
+            #    continue
+            #}
+
+            if ($PreviousRG -ne $VM.ResourceGroupName)
+            {
+                Write-Host "`tResourceGroup: " $VM.ResourceGroupName
+                $PreviousRG = $VM.ResourceGroupName
+            }
+
+            if ($VM.PowerState -ne "VM running")
+            {
+                Write-Host -ForegroundColor Gray `t`t $VM.Name " in resource group : VM is not running"
+                continue
+            }
+        
+            if ($VM.StorageProfile.OsDisk.OsType.ToString() -ne "Linux")
+            {
+                Write-Host -ForegroundColor Gray `t`t $VM.Name ": VM is not running Linux OS"
+                continue
+            }
+
+            $check = Invoke-AzVMRunCommand -ResourceGroupName $vm.ResourceGroupName -Name $vm.Name -CommandId 'RunShellScript' -ScriptPath $checkScriptPath
+ 
+            $split = $check.Value.Message.Split(" ",[System.StringSplitOptions]::RemoveEmptyEntries)
+            if ($check.Value.Message.Contains("/opt/omi/bin/omiserver:"))
+            {
+                $pkgVer = $split[3]
+                if($pkgVer -eq "OMI-1.6.8-1")

```

Comment:
fine for short term but later on if OMI releases newer, this will give false positives due to exact match

Reply:
yes, this is OK for now. We'll revise for future use..